internal/librarian/state.go

```diff
@@ -158,17 +158,24 @@ func saveLibrarianState(repoDir string, state *config.LibrarianState) error {
 	return os.WriteFile(path, bytes, 0644)
 }
 
-// readLibraryState reads the library state from a container response.
+// readLibraryState reads the library state from a container response, if it exists.
+// If the response file does not exist, readLibraryState succeeds but returns a nil pointer.
 //
 // The response file is removed afterwards.
 func readLibraryState(jsonFilePath string) (*config.LibraryState, error) {
 	data, err := os.ReadFile(jsonFilePath)
 	defer func() {
-		if err := os.Remove(jsonFilePath); err != nil {
+		if err := os.Remove(jsonFilePath); err != nil && !errors.Is(err, os.ErrNotExist) {
 			slog.Warn("fail to remove file", slog.String("name", jsonFilePath), slog.Any("err", err))
 		}
 	}()
 	if err != nil {
+		// If we only failed to read the file because it didn't exist, just succeed
+		// with a nil pointer.
+
+		if errors.Is(err, os.ErrNotExist) {
```

Comment:
I think it's better we check the stat of the file and returns early if it doesn't exist.

Reply:
I did wonder about that... @codyoss do you have a preference here?

Reply:
Ping @codyoss for a decision on this? (Would really like to get this PR in, as it will stop Go generation from failing post-container.)

Reply:
I am no preference. I think what is here is fine.

Reply:
Having chatted with Cody, I'll keep this as it is. Even if if we were to use Stat, we'd still be catching an error.